electrum/util.py

```diff
@@ -1952,12 +1952,16 @@ async def open_connection(self, host=None, port=None, **kwargs):
         return reader, writer
 
     @classmethod
-    def from_proxy_dict(cls, proxy: dict = None) -> Optional['MySocksProxy']:
-        if not proxy:
+    def from_network_settings(cls, network: Optional['Network']) -> Optional['ESocksProxy']:
+        if not network or not network.proxy:
             return None
+        proxy = network.proxy
         username, pw = proxy.get('user'), proxy.get('password')
         if not username or not pw:
-            auth = None
+            if network.is_proxy_tor:
```

Comment:
could you add a comment/fixme here saying that `is_proxy_tor` is actually tri-state, and if it's value is None, then we don't yet know if it is Tor
This actually makes all this prone to timing (`is_proxy_tor` gets set to None on network settings changes, briefly).

Reply:
Done. It is briefly None only on proxy changes, not on other network changes.

not a FIXME, a tri-state here is actually useful, and one can hook the 'tor_probed' callback to defer actions until the tor probe is done.